File: rximagepicker_support_wechat/src/main/java/com/qingmei2/rximagepicker_extension_wechat/ui/WechatImagePickerActivity.java
package com.qingmei2.rximagepicker_extension_wechat.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.qingmei2.rximagepicker.entity.Result;
import com.qingmei2.rximagepicker.ui.ActivityPickerViewController;
import com.qingmei2.rximagepicker_extension.entity.SelectionSpec;
import com.qingmei2.rximagepicker_extension_wechat.R;

import io.reactivex.functions.Action;
import io.reactivex.functions.Consumer;

@SuppressWarnings("CheckResult")
public class WechatImagePickerActivity extends AppCompatActivity {

    public static final int REQUEST_CODE_PREVIEW = 23;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        setTheme(SelectionSpec.getInstance().themeId);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_picker_wechat);
        displayPickerView();
    }

    private void displayPickerView() {
        WechatImagePickerFragment fragment = new WechatImagePickerFragment();
        getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.fl_container, fragment)
                .commit();

        fragment.pickImage()
                .subscribe(new Consumer<Result>() {
                    @Override
                    public void accept(Result result) throws Exception {
                        ActivityPickerViewController.getInstance().emitResult(result);
                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        ActivityPickerViewController.getInstance().emitError(throwable);
                    }
                }, new Action() {
                    @Override
                    public void run() throws Exception {
                        closure();
                    }
                });
    }

    public void closure() {
        ActivityPickerViewController.getInstance().endResultEmitAndReset();
        finish();
    }

    @Override
    public void onBackPressed() {
        closure();
    }
}

File: rximagepicker_support_wechat/src/main/java/com/qingmei2/rximagepicker_extension_wechat/ui/WechatImagePickerActivity.kt
package com.qingmei2.rximagepicker_extension_wechat.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.qingmei2.rximagepicker.ui.ActivityPickerViewController
import com.qingmei2.rximagepicker_extension.entity.SelectionSpec
import com.qingmei2.rximagepicker_extension_wechat.R

class WechatImagePickerActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(SelectionSpec.instance!!.themeId)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_picker_wechat)
        displayPickerView()
    }

    private fun displayPickerView() {
        val fragment = WechatImagePickerFragment()
        supportFragmentManager
                .beginTransaction()
                .add(R.id.fl_container, fragment)
                .commit()

        fragment.pickImage()
                .subscribe({ result -> ActivityPickerViewController.instance.emitResult(result) },
                        { throwable -> ActivityPickerViewController.instance.emitError(throwable) },
                        { closure() })
    }

    fun closure() {
        ActivityPickerViewController.instance.endResultEmitAndReset()
        finish()
    }

    override fun onBackPressed() {
        closure()
    }

    companion object {

        const val REQUEST_CODE_PREVIEW = 23
    }
}
